mlir/utils/vim/syntax/cir.vim

```diff
@@ -0,0 +1,124 @@
+" Vim syntax file
+" Language:   cir
+" Maintainer: The CIR team
+" Version:      $Revision$
+" Some parts adapted from the LLVM and MLIR vim syntax file.
+
+if version < 600
+  syntax clear
+elseif exists("b:current_syntax")
+  finish
+endif
+
+syn case match
+
+" Types.
+"
+syn keyword mlirType index f16 f32 f64 bf16
```

Comment:
`syn keyword mlirType index f16 f32 f64 bf16`

I think we can create a matcher for FP because there is f128, f16, we can add them for now and later create a regex for them